controllers/druid/handler.go

```diff
@@ -391,6 +395,158 @@ func checkCrashStatus(sdk client.Client, m *v1alpha1.Druid) {
+
+	sts := makeStatefulSetEmptyObj()
+	// During creation/initalization of druid cluster, this will fail since the sts will not be present
+	// So logging this as info, event type normal, subsequent reconciles should work fine.
+	if err := sdk.Get(context.TODO(), *namespacedName(nodeSpecUniqueStr, drd.Namespace), sts); err != nil {
+		msg := fmt.Sprintf("failed to get [StatefuleSet:%s] due to [%s]", nodeSpecUniqueStr, err.Error())
+		logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+		sendEvent(sdk, drd, v1.EventTypeNormal, "GET_FAIL", err.Error())
+		return
+	}
+
+	// ignoring error, function shall log and sendEvent for it
+	pvc := getPVCList(sdk, nodeSpec, drd, nodeSpecUniqueStr)
+
+	desVolumeClaimTemplateSize, currVolumeClaimTemplateSize, pvcSize := getVolumeClaimTemplateSizes(sts, nodeSpec, pvc)
+
+	ok := checkIfStsHasEverythingElseSame(sdk, drd, ls, configMapSHA, serviceName, nodeSpecUniqueStr, nodeSpec)
+	if !ok {
+		return
+	}
+
+	// Validate Request, shrinking of pvc not supported
+	// desired size cant be less than current size
+	// in that case re-create sts/pvc which is a user execute manual step
+	if desVolumeClaimTemplateSize.String() < currVolumeClaimTemplateSize.String() {
+		msg := fmt.Sprintf("Request for Shrinking of sts pvc size [sts:%s] in [namespace:%s] is not Supported", sts.Name, sts.Namespace)
+		logger.Info(msg)
+		sendEvent(sdk, drd, v1.EventTypeWarning, "STS_SIZE_FAIL", msg)
+		return
+	}
+
+	// In case size dont match and dessize > currsize, delete the sts using casacde=false / propagation policy set to orphan
+	// The operator on next reconcile shall create the sts with latest changes
+	if desVolumeClaimTemplateSize.String() != currVolumeClaimTemplateSize.String() {
+		msg := fmt.Sprintf("Detected Change in VolumeClaimTemplate Sizes for Statefuleset [%s] in Namespace [%s], desVolumeClaimTemplateSize: [%s], currVolumeClaimTemplateSize: [%s]\n, deleteing STS [%s] with casacde=false]", sts.Name, sts.Namespace, desVolumeClaimTemplateSize.String(), currVolumeClaimTemplateSize.String(), sts.Name)
+		logger.Info(msg)
+		sendEvent(sdk, drd, v1.EventTypeNormal, "STS_SIZE_DETECTED", msg)
+
+		if err := sdk.Delete(context.TODO(), sts, client.PropagationPolicy(metav1.DeletePropagationOrphan)); err != nil {
+			e := fmt.Errorf("failed to Delete [StatefuleSet:%s] due to [%s]", nodeSpecUniqueStr, err.Error())
+			logger.Error(e, e.Error(), "name", drd.Name, "namespace", drd.Namespace)
+			sendEvent(sdk, drd, v1.EventTypeWarning, "DELETE_FAIL", e.Error())
+		} else {
+			msg := fmt.Sprintf("[StatefuleSet:%s] successfully deleted with casacde=false", sts.Name)
+			logger.Info(msg, "name", drd.Name, "namespace", drd.Namespace)
+			sendEvent(sdk, drd, v1.EventTypeNormal, "DELETE_SUCCESS", msg)
+		}
+	}
+
+	// In case size dont match, patch the pvc with the desiredsize from druid CR
+	if desVolumeClaimTemplateSize.String() != pvcSize.String() {
```

Comment:
what happens if we just recreated the sts with new size in pvc and did not explicitly patched the PVCs  i.e. what happens if we removed this whole if clause altogether?

Reply:
Not patching pvc will still result in the same size as earlier size. In case of no patch, sts volumeclaimtemplate sizes  and pvc sizes will differ.